src/main/java/org/lanternpowered/server/resource/LanternResourcePathBuilder.java

```diff
@@ -0,0 +1,101 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.lanternpowered.server.resource;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static org.lanternpowered.server.resource.LanternResourcePath.NAMESPACE_PATH_PATTERN;
+import static org.lanternpowered.server.resource.LanternResourcePath.NAMESPACE_PATTERN;
+import static org.lanternpowered.server.resource.LanternResourcePath.PATH_PATTERN;
+import static org.lanternpowered.server.util.Conditions.checkPlugin;
+
+import org.spongepowered.api.resource.ResourcePath;
+
+import java.util.regex.Matcher;
+
+@SuppressWarnings({"ConstantConditions", "NullableProblems"})
+public final class LanternResourcePathBuilder implements ResourcePath.Builder {
+
+    private String namespace;
+    private String path;
+
+    @Override
+    public ResourcePath.Builder namespace(String namespace) throws IllegalArgumentException {
+        checkNotNull(namespace, "namespace");
+        checkState(NAMESPACE_PATTERN.matcher(namespace).matches(),
+                "%s doesn't match the namespace pattern: %s", namespace, NAMESPACE_PATTERN.pattern());
+        this.namespace = namespace;
+        return this;
+    }
+
+    @Override
+    public ResourcePath.Builder plugin(Object plugin) {
+        this.namespace = checkPlugin(plugin, "plugin").getId();
+        return this;
+    }
+
+    @Override
+    public ResourcePath.Builder path(String path) throws IllegalArgumentException {
+        checkNotNull(path, "path");
+        final Matcher matcher = PATH_PATTERN.matcher(path);
+        checkState(matcher.matches(),
+                "%s doesn't match the path pattern: %s", path, PATH_PATTERN.pattern());
+        this.namespace = matcher.group(1);
+        return this;
+    }
+
+    @Override
+    public ResourcePath.Builder parse(String path) throws IllegalArgumentException {
+        checkNotNull(path, "path");
+        final Matcher matcher = NAMESPACE_PATH_PATTERN.matcher(path);
+        checkState(matcher.matches(),
+                "%s doesn't match the namespaced path pattern: %s", path, NAMESPACE_PATH_PATTERN.pattern());
+        this.namespace = matcher.group(1);
```

Comment:
How does this handle a non-existent namespace?

Reply:
Non existent would be `path/to/file`, wildcard namespace `:path/to/file`

Reply:
I haven't currently implemented `path/to/file` support